statescript/executor.go

```diff
@@ -179,8 +210,43 @@ func execute(name string, timeout time.Duration) int {
 	return 0
 }
 
+// Catches a script that requests a retry in a loop. Is limited by the total window given to a script demanding a
+// retry.
+func executeScript(s os.FileInfo, dir string, l Launcher, timeout time.Duration, ignoreError bool) error {
+
+	iet := time.Now()
+	for {
+		switch ret := execute(filepath.Join(dir, s.Name()), timeout); ret {
+		case 0:
+			// success
+			return nil
+		case exitRetryLater:
```

Comment:
I think the logic here is wrong. We should first sleep and then ignore errors if any. 
```
case exitRetryLater:
  if time.Since(iet) <= l.getRetryTimeout() {
    log.Errorf("statescript: %s requested a retry", s.Name())
    time.Sleep(l.getRetryInterval())
    continue
  }
  else if ignoreError {
    log.Errorf("statescript: ignoring error executing '%s': %d", s.Name(), ret)
    return nil
  }
  return errors.New("statescript: retry time-limit exceeded")
```